### PR TITLE
extended dims cube

### PR DESCRIPTION
**Pull request by atolopko-czi**
For https://github.com/chanzuckerberg/cellxgene-census/issues/603

### CODE REVIEW

File: memento/cell_census_summary_cube.py [hunk 1]
```diff
@@ -30,53 +30,49 @@
...
     "dataset_id",
+]
+
+CUBE_TILEDB_ATTRS_OBS = [
     "assay",
     "suspension_type",
     "donor_id",
     "disease",
     "sex"
 ]
```
Comment (atolopko-czi) [inline memento/cell_census_summary_cube.py:48]: @mincheoly The above split of columns between "dims" and "attrs" allows us to select the columns that TileDB will index (in the sense of a relational database), allowing for efficient filtering on these columns. What is the access pattern for the cube? If there are 7 columns, will filtering always be performed on all columns? Or is it arbitrary?

**Comment by atolopko-czi** [comment]
@mincheoly This code is currently the performance hotspot:
```
gene_rvs = gen.multinomial(n_obs, counts / counts.sum(), size=num_boot).T
```
With the extra dimensions, this is being called considerably more times. The new cube will have ~102970 unique dimension tuples vs the original cube's ~4047, which means 25x more `multinomial` calls.

Is there a faster implementation we can find?

Can we possibly decrease `num_boot`, as this helps it run faster?

**Comment by mincheoly** [comment]
i think reducing num_boot to ~1000 (maybe even 500) should be fine (and we can compare later on a smaller cube?)

I'm surprised that's the bottleneck and not getting the unique counts. i remember testing a few different ways to generate the multinomial and they didnt seem hugely different - but maybe things have changed since (like https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.multinomial.html)

Would reducing the num_boot get it to a point where its runnable?

**Comment by atolopko-czi** [comment]
> i think reducing num_boot to ~1000 (maybe even 500) should be fine (and we can compare later on a smaller cube?)

Great! I'll start with 500 and see that affects performance.

> Would reducing the num_boot get it to a point where its runnable?

It should at least make that one call 10x faster, and since it's the performance hotspot, that may be enough. Stay tuned!

**Comment by atolopko-czi** [comment]
@mincheoly (cc: @pablo-gar) Since we can generate the cube in ~1 day, perhaps that is sufficient for now? I figure that as long as you can get the the larger dimension cube generated once, you can experiment with it. In which case, I'll pause on doing further optimization work. And if we get to the point of needing to generate this cube on a regular basis (e.g. for every Census data release), I will put more effort into optimizing.

I think the code changes are worth merging to main now, as is, but I'll wait for your approval on this PR before merging (no rush). Thanks!

**Pull request merged**